packages/plugins/table/src/index.tsx

```diff
@@ -2,26 +2,27 @@ import { StatefulPlugin, string } from '@edtr-io/plugin'
 import * as React from 'react'
 import ReactMarkdown from 'react-markdown'
 
-import { createTableEditor } from './editor'
+import { TableEditor } from './editor'
 
 export const tableState = string()
 
 export function createTablePlugin(
-  config: TablePluginConfig
-): StatefulPlugin<typeof tableState> {
+  renderMarkdown = defaultRenderMarkdown
+): StatefulPlugin<typeof tableState, TablePluginConfig> {
   return {
-    Component: createTableEditor(config),
+    Component: TableEditor,
+    config: {
```

Comment:
A plugin exposes its config (even if it's empty, we need to pass `{}`, otherwise the typing gets really annoying). Here, it should use the stuff passed to the factory.

If it uses a theme, it may also provide a config factory of the form (see https://github.com/edtr-io/edtr-io/commit/61eb0349dc4f9604231894a228e37c30a0051461#diff-b198ccead7220120e13365fe3b890deeR12-R36 as an example)
```.ts
function createConfig(theme: Theme): FooPluginConfig {
  return {
    bar: theme.editor.whatever
  }
}
```